lib/proxy/class-iputils.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+/*
+ * This file was borrowed and modified from this Symfony package:
+ * https://github.com/symfony/http-foundation
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please see https://github.com/symfony/http-foundation/blob/master/LICENSE
+ */
+
+namespace Automattic\VIP\Proxy;
+
+/**
+ * Http utility functions.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class IpUtils {
+
+	/**
+	 * This class should not be instantiated.
+	 */
+	private function __construct() {
+	}
+
+	/**
+	 * Checks if an IPv4 or IPv6 address is contained in the list of given IPs or subnets.
+	 *
+	 * @param string       $request_ip IP to check
+	 * @param string|array $ips       List of IPs or subnets (can be a string if only a single one)
+	 *
+	 * @return bool Whether the IP is valid
+	 */
+	// phpcs:ignore WordPress.NamingConventions.ValidFunctionName.MethodNameInvalid
```

Comment:
The insertion here techincally breaks the association of the DocBlock above with the method.

Could consider moving the ignore line to the end of the next line instead.

Also, how widely used is the method name? Could it be renamed / new `check_ip()` method added and have this one point to it, etc?

Reply:
Turns out not that many:) I changed the name now.

Reply:
As these were public methods, have you checked across the customer code bases to see if anyone is using the old method names?

Reply:
I've grepped through all client site repos and I haven't seen any usage of this class in `mu-plugins`, so the changed should be safe

Reply:
Thanks Suso! To add to this all the requires of this file are scoped within functions, so the client would have to require the code themselv.

Reply:
Great, thanks for checking 👍🏻